src/vxworks/mod.rs

```diff
@@ -220,18 +220,12 @@ s! {
                          pub st_reserved4 : ::c_int,
     }
 
-    //b_struct__Timespec.h
-    pub struct _Timespec {
-        pub tv_sec  : ::time_t,
-        pub tv_nsec : ::c_long,
-    }
-
     // b_struct__Sched_param.h
-    pub struct _Sched_param {
+    pub struct sched_param {
```

Comment:
I want to still run the libc test on this. Can we keep the old code to define _Sched_param but define sched_param as an alias for _Sched_param? Then maybe just skip testing of sched_param, but still test the "real" struct _Sched_param.

Reply:
It seems the libc-test can't handle the type define right now, with
pub type sched_param = _Sched_param;

I am seeing this error:

running: "wr-c++" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-static" "-Wall" "-We\
xtra" "-Wall" "-Wextra" "-Werror" "-Wno-unused-parameter" "-Wno-type-limits" "-Wno-address-of-packed-member" "-Wno-unknown-warning-o\
ption" "-Wno-deprecated-declarations" "-o" "/folk/prj-rust-dev/bpang/V7LIBC-1006/dev/libc/target/x86_64-wrs-vxworks/debug/build/libc\
-test-b2a25489b4cb1bf6/out/main.o" "-c" "/folk/prj-rust-dev/bpang/V7LIBC-1006/dev/libc/target/x86_64-wrs-vxworks/debug/build/libc-te\
st-b2a25489b4cb1bf6/out/main.c"
cargo:warning=/folk/prj-rust-dev/bpang/V7LIBC-1006/dev/libc/target/x86_64-wrs-vxworks/debug/build/libc-test-b2a25489b4cb1bf6/out/mai\
n.c:4456:69: error: use of undeclared identifier 'sched_param'
cargo:warning=             uint64_t __test_size_sched_param(void) { return sizeof(sched_param); }

Reply:
In VxWorks head files, I see both ''struct _Sched_param" and "struct sched_param" are defined, and they are exactly same for user space. What confused me is in some place we use "_Sched_param", but in some place we use "sched_param". Why we do it in such way?

Reply:
I think it is to do with avoiding namespace pollution for PSE52 conformance.

See comment in http://stash.wrs.com/projects/VX7/repos/vxworks/browse/helix/guests/vxworks-7/pkgs_v2/os/core/user/h/base/b_pthread_attr_t.h?at=refs%2Fheads%2Fvx7-integration#36

```
* Note: we can't simply use the sched_param stuctures here because this                                                                                                                                    
 * conflicts with the namespace isolation rules when users of the                                                                                                                                           
 * pthread_attr_t structure type other than pthread.h (such as signal.h)                                                                                                                                    
 * are included directly or indirectly by an application.
```

Reply:
" error: use of undeclared identifier 'sched_param'"

Hmm. Maybe you need to add another header file to to the libc test suite to provide sched_param? I think we specify which headers to include, right?

Reply:
The head file for sched_param is already included, to avoid the error, we need to make the generated code as "sizeof(struct sched_param)" istead of "sizeof(sched_param)". It can be done through a little trick in libc-test/build.rs, but do we have to do it? from my opinion, using "sched_param" is exactly same as "_Sched_param" in libc.

Reply:
OK maybe I misunderstood: in your original code, we are skipping testing threadAttrSchedparam, but are we testing "sched_param"? If so, then I'm fine with this code.

Reply:
Yes, in my original code, we are skipping testing threadAttrSchedparam as its type is "_Sched_param" in VxWorks, and we are testing "sched_param".